bazel/envoy_build_system.bzl

```diff
@@ -0,0 +1,47 @@
+# References to Envoy external dependencies should be wrapped with this function.
+def envoy_external_dep_path(dep):
+    return "//external:%s" % dep
+
+ENVOY_COPTS = [
+    "-fno-omit-frame-pointer",
+    "-Wall",
+    "-Wextra",
+    "-Werror",
+    "-Wnon-virtual-dtor",
+    "-Woverloaded-virtual",
+    "-Wold-style-cast",
+    "-std=c++0x",
+    "-includesource/precompiled/precompiled.h",
+    "-iquoteinclude",
+    "-iquotesource",
+]
+
+# Envoy C++ library targets should be specified with this function.
+def envoy_cc_library(name,
+                     srcs = [],
+                     public_hdrs = [],
+                     hdrs = [],
+                     external_deps = [],
+                     deps = [],
+                     copts = []):
```

Comment:
you may want a **kwargs here to catch everything else that you don't touch and pass to cc_library (e.g. visibility, defines)